admin/app/components/solidus_admin/orders/cart/component.js

```diff
@@ -0,0 +1,129 @@
+import { Controller } from "@hotwired/stimulus"
+import { useClickOutside, useDebounce } from "stimulus-use"
+
+const LOADING = "Loading..."
+const INITIAL = "Type to search"
+const EMPTY = "No results"
+const QUERY_KEY = "q[name_or_variants_including_master_sku_cont]"
+
+export default class extends Controller {
+  static targets = ["result", "results", "searchField"]
+  static values = { results: String, productsUrl: String }
+  static debounces = [{
+    name: "requestSubmitForLineItems",
+    wait: 500,
+  }, "search"]
+
+  get query() {
+    return this.searchFieldTarget.value
+  }
+
+  get selectedResult() {
+    // Keep the index within boundaries
+    if (this.selectedIndex < 0) this.selectedIndex = 0
+    if (this.selectedIndex >= this.resultTargets.length) this.selectedIndex = this.resultTargets.length - 1
+
+    return this.resultTargets[this.selectedIndex]
+  }
+
+  connect() {
+    useClickOutside(this)
+    useDebounce(this)
+
+    this.selectedIndex = 0
+    this.lineItemsToBeSubmitted = []
+
+    if (this.query) {
+      this.showResults()
+      this.search()
+    }
+  }
+
+  selectResult(event) {
+    switch (event.key) {
+      case "Enter":
+        event.preventDefault()
+        this.selectedResult?.click()
+        break
+      case "ArrowUp":
+        event.preventDefault()
+        this.selectedIndex -= 1
+        this.render()
+        break
+      case "ArrowDown":
+        event.preventDefault()
+        this.selectedIndex += 1
+        this.render()
+        break
+    }
+  }
+
+  clickOutside() {
+    this.openResults = false
+    this.render()
+  }
+
+  async search() {
+    const query = this.query
+
+    if (query) {
+      this.resultsValue = LOADING
+      this.render()
+
+      this.resultsValue = (await (await fetch(`${this.productsUrlValue}?${QUERY_KEY}=${query}`)).text()) || EMPTY
```

Comment:
Is this query configurable somewhere? We need to be able to change it as we are in the current variant select

Reply:
Currently no, but this is not the only place in which we'll have to fetch products/variants, once we bump into the next one we'll surely get out something more generic and possibly have a `ui/…` component rather than one specific for the `orders/…` pages.

Reply:
Makes sense.